xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -1247,6 +1248,33 @@ int CGUIWindowManager::GetActiveWindow() const
   return WINDOW_INVALID;
 }
 
+int CGUIWindowManager::GetActiveWindowID()
+{
+  // Get the currently active window
+  int iWin = GetActiveWindow() & WINDOW_ID_MASK;
+
+  // If there is a dialog active get the dialog id instead
+  if (HasModalDialog())
+    iWin = GetTopMostModalDialogID() & WINDOW_ID_MASK;
```

Comment:
I'm trying to avoid changing the code more than necessary when yanking things out of CApplication to avoid introducing bugs and to avoid having functional changes hidden in moved blocks. I'm not opposed to changing it if there's a gain to be had though